### PR TITLE
[SPARK-28287][SQL][PYTHON][TESTS] Convert and port 'udaf.sql' into UDF test base

### PR DESCRIPTION
**Pull request by vinodkc**
## What changes were proposed in this pull request?

This PR adds some tests converted from 'udaf.sql' to test UDFs

<details><summary>Diff comparing to 'udaf.sql'</summary>
<p>

```diff
diff --git a/sql/core/src/test/resources/sql-tests/results/udaf.sql.out b/sql/core/src/test/resources/sql-tests/results/udf/udf-udaf.sql.out
index f4455bb717..e1165931d6 100644
--- a/sql/core/src/test/resources/sql-tests/results/udaf.sql.out
+++ b/sql/core/src/test/resources/sql-tests/results/udf/udf-udaf.sql.out
@@ -21,7 +21,7 @@ struct<>


 -- !query 2
-SELECT default.myDoubleAvg(int_col1) as my_avg from t1
+SELECT default.myDoubleAvg(udf(int_col1)) as my_avg from t1
 -- !query 2 schema
 struct<my_avg:double>
 -- !query 2 output
@@ -29,7 +29,7 @@ struct<my_avg:double>


 -- !query 3
-SELECT default.myDoubleAvg(int_col1, 3) as my_avg from t1
+SELECT default.myDoubleAvg(udf(int_col1), udf(3)) as my_avg from t1
 -- !query 3 schema
 struct<>
 -- !query 3 output
@@ -46,7 +46,7 @@ struct<>


 -- !query 5
-SELECT default.udaf1(int_col1) as udaf1 from t1
+SELECT default.udaf1(udf(int_col1)) as udaf1 from t1
 -- !query 5 schema
 struct<>
 -- !query 5 output
```

</p>
</details>

## How was this patch tested?

Tested as guided in SPARK-27921.

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
@vinodkc,

```diff
 -- !query 2
-SELECT default.myDoubleAvg(int_col1) as my_avg from t1
+SELECT default.myDoubleAvg(int_col1), default.myDoubleAvg(udf(int_col1)) as my_avg from t1
 -- !query 2 schema
-struct<my_avg:double>
+struct<mydoubleavg(CAST(int_col1 AS DOUBLE)):double,my_avg:double>
 -- !query 2 output
-102.5
+102.5  102.5
```

This one looks a bit weird comparing to the original file.

**Comment by vinodkc** [comment]
> @vinodkc,
> 
> ```diff
>  -- !query 2
> -SELECT default.myDoubleAvg(int_col1) as my_avg from t1
> +SELECT default.myDoubleAvg(int_col1), default.myDoubleAvg(udf(int_col1)) as my_avg from t1
>  -- !query 2 schema
> -struct<my_avg:double>
> +struct<mydoubleavg(CAST(int_col1 AS DOUBLE)):double,my_avg:double>
>  -- !query 2 output
> -102.5
> +102.5  102.5
> ```
> 
> This one looks a bit weird comparing to the original file.

Fixed the review comment, thanks

**Comment by HyukjinKwon** [comment]
@vinodkc, https://github.com/apache/spark/pull/25130 is merged. Can you rebase and update?

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/core/src/test/resources/sql-tests/inputs/udf/udf-udaf.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+-- This test file was converted from udaf.sql.
+
+CREATE OR REPLACE TEMPORARY VIEW t1 AS SELECT * FROM VALUES
+(1), (2), (3), (4)
+as t1(int_col1);
+
+CREATE FUNCTION myDoubleAvg AS 'test.org.apache.spark.sql.MyDoubleAvg';
+
+SELECT default.myDoubleAvg(udf(int_col1)) as my_avg from t1;
```
Comment (HyukjinKwon) [inline sql/core/src/test/resources/sql-tests/inputs/udf/udf-udaf.sql]: @vinodkc, let's add a different combination in general. For instance,

```
udf(default.myDoubleAvg(udf(int_col1)))
```

```
udf(default.myDoubleAvg(int_col1))
```

**Comment by HyukjinKwon** [comment]
Looks good to me otherwise.

**Comment by vinodkc** [comment]
Closing this PR due to code synching will raise a new PR

**Pull request closed without merging**